### PR TITLE
[Impeller] remove incremental allocation during filled path tessellation.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Don't allocate a polyline (and resulting std::vectors) when performing convex tessellation. Instead use a listener pattern and iterate through path verbs.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I don't think the goldens run till the sub-builds are done.

**Comment by jonahwilliams** [comment]
It turns out they actually do, you can click into the flutter gold pending task and get a link:

https://flutter-engine-gold.skia.org/search?issue=52401&crs=github&patchsets=2&corpus=flutter-engine

This still has some bugs :)

**Comment by jonahwilliams** [comment]
Working now! modulo CI issues

File: impeller/geometry/geometry_benchmarks.cc [hunk 1]
```diff
@@ -44,39 +43,22 @@ template <class... Args>
...
-          // NOLINTNEXTLINE(clang-analyzer-cplusplus.Move)
-          1.0f, std::move(points),
-          [&points](Path::Polyline::PointBufferPtr reclaimed) {
-            points = std::move(reclaimed);
-          });
-      single_point_count = polyline.points->size();
-      point_count += single_point_count;
-    }
+    auto polyline = path.CreatePolyline(
+        // Clang-tidy doesn't know that the points get moved back before
```
Comment (chinmaygarde) [inline impeller/geometry/geometry_benchmarks.cc:53]: Oof, this is hard for me to follow lol. Thanks for the comment.

**Review by chinmaygarde — APPROVED** [review]

**Comment by gaaclarke** [comment]
This was about a 10% reduction in `90th_percentile_frame_rasterizer_time_millis:	static_path_tessellation_perf__timeline_summary`.

https://flutter-flutter-perf.skia.org/e/?keys=X2a72c135a55da5cfa04230cc0ad78045&selected=commit%3D40603%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DPixel_7_Pro%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253D90th_percentile_frame_rasterizer_time_millis%252Ctest%253Dstatic_path_tessellation_perf__timeline_summary%252C&xbaroffset=40603

**Pull request merged**